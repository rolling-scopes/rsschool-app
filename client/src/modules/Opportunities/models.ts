import { ResumeDtoEnglishLevelEnum, ResumeDtoMilitaryServiceEnum } from 'api';
import { Dayjs } from 'dayjs';

export interface CourseData {
  locationName: string;
  courseFullName: string;
  isExpelled: boolean;
  certificateId: string | null;
  isCourseCompleted: boolean;
  totalScore: number;
  rank: number | null;
  mentor: {
    githubId: string;
    name: string;
  };
}

export type VisibleCourses = number[];

export interface CourseDataShortened {
  courseId: number;
  courseFullName: string;
}

export interface VisibleCoursesFormData {
  [id: string]: boolean;
}

export type ContactType = 'phone' | 'email' | 'skype' | 'telegram' | 'linkedin' | 'githubUsername' | 'website';

export interface UserData {
  avatarLink: string | null;
  name: string | null;
  desiredPosition: string | null;
  selfIntroLink: string | null;
  englishLevel: ResumeDtoEnglishLevelEnum | null;
  militaryService: ResumeDtoMilitaryServiceEnum | null;
  notes: string | null;
  startFrom: string | null;
  fullTime: boolean;
  locations: string | null;
}

export type Contacts = {
  [key in ContactType]: string | null;
};

export interface FieldData {
  name: string[];
  value: any;
  touched: boolean;
  validating: boolean;
  errors: string[];
}

export interface AllUserCVData extends Omit<UserData, 'uuid'>, Contacts {
  visibleCourses: VisibleCourses;
}

export interface UserDataToSubmit extends Omit<UserData, 'startFrom'> {
  startFrom: Dayjs;
  visibleCourses: VisibleCourses;
}

export interface AllDataToSubmit extends UserDataToSubmit, Contacts {}

export interface EditCVData extends AllUserCVData {
  expires: number | null;
  courses: CourseDataShortened[];
  visibleCourses: VisibleCourses;
}

export interface GetFullCVData extends AllUserCVData {
  expires: number | null;
  courses: CVStudentStats[];
  feedback: CVFeedback[];
}

export interface CVStudentStats {
  locationName: string;
  courseFullName: string;
  certificateId: string | null;
  isCourseCompleted: boolean;
  totalScore: number;
  rank: number | null;
  mentor: {
    githubId: string;
    name: string;
  };
}

export interface JobSeekerStudentStats extends CVStudentStats {
  courseName: string;
}

export interface JobSeekerData {
  name: string | null;
  desiredPosition: string | null;
  githubId: string;
  englishLevel: ResumeDtoEnglishLevelEnum | null;
  fullTime: boolean;
  locations: string | null;
  startFrom: string | null;
  courses: JobSeekerStudentStats[];
  feedback: JobSeekerFeedback[];
  expires: number;
  isHidden: boolean;
}

export interface CVFeedback {
  comment: string;
  feedbackDate: string;
}

export interface JobSeekerFeedback {
  badgeId: string;
}
