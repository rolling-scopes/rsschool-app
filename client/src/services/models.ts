import { Session } from 'components/withSession';
import {
  StudentBasic as CommonStudentBasic,
  DiscordServer,
  UserGroup,
  InterviewQuestion,
  InterviewQuestionCategory,
} from 'common/models';
import type { ProfileCourseDto } from 'api';

export interface Course extends ProfileCourseDto {}
export interface UpdateCourse extends ProfileCourseDto {
  disciplineId: number | null;
}

export interface StudentBasic extends CommonStudentBasic {}
export type { DiscordServer };
export type { UserGroup };
export type { InterviewQuestion };
export type { InterviewQuestionCategory };

export interface Mentor {
  lastName: string;
  firstName: string;
  githubId: string;

  studentId: number;
  userId: number;
  courseId: number;
}

export interface PageWithModalState<T> {
  data: T[];
  modalData: Partial<T> | null;
  modalAction: 'update' | 'create';
}

export interface CoursePageProps {
  session: Session;
  course: Course;
  params: Record<string, string>;
}

export type CourseOnlyPageProps = {
  course: Course;
  params?: Record<string, string>;
};
