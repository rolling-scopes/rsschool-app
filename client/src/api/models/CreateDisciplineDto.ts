/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type CreateDisciplineDto = {
    name: string;
}
