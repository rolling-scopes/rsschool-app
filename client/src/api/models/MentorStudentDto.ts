/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type MentorStudentDto = {
    name: string;
    id: number;
    cityName: string;
    countryName: string;
    githubId: string;
    totalScore: number;
    rank: number;
    hasFeedback: boolean;
}
