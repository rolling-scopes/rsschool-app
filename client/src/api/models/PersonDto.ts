/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type PersonDto = {
    name: string;
    id: number;
}
