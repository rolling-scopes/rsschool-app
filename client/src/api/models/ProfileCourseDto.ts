/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type ProfileCourseDto = {
    id: number;
    name: string;
    alias: string;
}
