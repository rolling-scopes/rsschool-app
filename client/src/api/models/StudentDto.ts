/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type StudentDto = {
    name: string;
    id: number;
    cityName: string;
    countryName: string;
    githubId: string;
    totalScore: number;
    rank: number;
}
