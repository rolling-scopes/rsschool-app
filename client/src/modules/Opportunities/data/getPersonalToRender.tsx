import { UserData, MilitaryServiceStatus } from 'common/models/cv';
import capitalize from 'lodash/capitalize';

export function getPersonalToRender(user: UserData) {
  const { selfIntroLink } = user;
  return [
    {
      title: 'English',
      value: user.englishLevel,
    },
    {
      title: 'Self Introduction',
      value: selfIntroLink ? (
        <a className="rs-link" target="_blank" rel="nofollow" href={selfIntroLink}>
          {selfIntroLink}
        </a>
      ) : (
        'Not Available'
      ),
    },
    {
      title: 'Ready to work full time',
      value: user.fullTime ? 'Yes' : 'No',
    },
    {
      title: 'Ready to work from',
      value: user.startFrom,
    },
    {
      title: 'Military service status',
      value: user.militaryService ? capitalize(militaryServiceDictionary[user.militaryService]) : 'Unknown',
    },
  ];
}

const militaryServiceDictionary: {
  [key in MilitaryServiceStatus]: string;
} = {
  served: 'served',
  notLiable: 'not liable',
  liable: 'liable',
};
