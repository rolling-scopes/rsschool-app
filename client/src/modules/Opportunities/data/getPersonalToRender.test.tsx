import { UserData } from '../models';
import { getPersonalToRender } from './getPersonalToRender';

describe('getPersonalToRender', () => {
  it('should return correct data in case all values are present', () => {
    const mockUserData = {
      englishLevel: 'B1',
      fullTime: true,
      locations: 'Minsk Belarus',
      militaryService: 'notLiable',
      selfIntroLink: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
      startFrom: '2022-09-24',
    } as unknown as UserData;

    const view = getPersonalToRender(mockUserData);
    expect(view).toMatchSnapshot();
  });

  it('should correctly handle missing values', () => {
    const mockUserData = {
      englishLevel: null,
      fullTime: false,
      locations: null,
      militaryService: null,
      selfIntroLink: null,
      startFrom: null,
    } as unknown as UserData;

    const view = getPersonalToRender(mockUserData);
    expect(view).toMatchSnapshot();
  });
});
