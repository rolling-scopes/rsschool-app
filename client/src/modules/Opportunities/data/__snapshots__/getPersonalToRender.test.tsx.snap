// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPersonalToRender should correctly handle missing values 1`] = `
[
  {
    "title": "English",
    "value": null,
  },
  {
    "title": "Self Introduction",
    "value": "Not Available",
  },
  {
    "title": "Ready to work full time",
    "value": "No",
  },
  {
    "title": "Ready to work from",
    "value": null,
  },
  {
    "title": "Military service status",
    "value": "Unknown",
  },
  {
    "title": "Locations",
    "value": "Unknown",
  },
]
`;

exports[`getPersonalToRender should return correct data in case all values are present 1`] = `
[
  {
    "title": "English",
    "value": "B1",
  },
  {
    "title": "Self Introduction",
    "value": <a
      className="rs-link"
      href="https://www.youtube.com/watch?v=dQw4w9WgXcQ"
      rel="nofollow"
      target="_blank"
    >
      https://www.youtube.com/watch?v=dQw4w9WgXcQ
    </a>,
  },
  {
    "title": "Ready to work full time",
    "value": "Yes",
  },
  {
    "title": "Ready to work from",
    "value": "2022-09-24",
  },
  {
    "title": "Military service status",
    "value": "Not liable",
  },
  {
    "title": "Locations",
    "value": "Minsk Belarus",
  },
]
`;
